src/screens/HomeScreen.js

```diff
@@ -146,104 +231,65 @@ export const HomeScreen = ({ navigation, route }) => {
+            placeholder: <NewsSectionPlaceholder navigation={navigation} title={categoryTitle} />,
+            query,
+            queryVariables: { ...queryVariables, categoryId },
+            showVolunteerEvents,
+            title: categoryTitle,
+            titleDetail: categoryTitleDetail
+          }}
         />
+      )
+    );
+  }
 
-        <Widgets widgetConfigs={widgetConfigs} />
-
-        {showNews &&
-          categoriesNews.map(
-            (
-              {
-                categoryButton,
-                categoryId,
-                categoryTitle,
-                categoryTitleDetail,
-                indexCategoryIds,
-                rootRouteName
-              },
-              index
-            ) => (
-              <HomeSection
-                key={index}
-                buttonTitle={categoryButton}
-                categoryId={categoryId}
-                title={categoryTitle}
-                titleDetail={categoryTitleDetail}
-                fetchPolicy={fetchPolicy}
-                navigate={() =>
-                  navigation.navigate(
-                    NAVIGATION.NEWS_ITEMS_INDEX({
-                      categoryId,
-                      categoryTitle,
-                      categoryTitleDetail,
-                      indexCategoryIds,
-                      rootRouteName
-                    })
-                  )
-                }
-                navigation={navigation}
-                placeholder={
-                  <NewsSectionPlaceholder navigation={navigation} title={categoryTitle} />
-                }
-                query={QUERY_TYPES.NEWS_ITEMS}
-                queryVariables={{ limit: 3, categoryId, excludeDataProviderIds }}
-              />
-            )
-          )}
-
-        {showPointsOfInterestAndTours && (
-          <HomeSection
-            buttonTitle={buttonPointsOfInterestAndTours}
-            title={headlinePointsOfInterestAndTours}
-            fetchPolicy={fetchPolicy}
-            navigate={() => navigation.navigate(NAVIGATION.CATEGORIES_INDEX)}
-            navigation={navigation}
-            query={QUERY_TYPES.POINTS_OF_INTEREST_AND_TOURS}
-            queryVariables={{ limit: 10, orderPoi: 'RAND', orderTour: 'RAND' }}
-          />
-        )}
-
-        {showEvents && (
-          <HomeSection
-            buttonTitle={buttonEvents}
-            title={headlineEvents}
-            fetchPolicy={fetchPolicy}
-            navigate={() => navigation.navigate(NAVIGATION.EVENT_RECORDS_INDEX)}
-            navigation={navigation}
-            query={QUERY_TYPES.EVENT_RECORDS}
-            queryVariables={{ limit: 3, order: 'listDate_ASC' }}
-            showVolunteerEvents={showVolunteerEvents}
-          />
-        )}
-
-        {route.params?.isDrawer && (
-          <>
-            <HomeService />
-            <About navigation={navigation} withHomeRefresh />
-          </>
-        )}
-      </ScrollView>
-    </SafeAreaViewFlex>
+  return (
+    isShow && (
```

Comment:
this is the second `isShow` check. what about having an early return of `null` when it is not true to avoid checking it multiple times?

Reply:
I never thought of that, but yes, I should add that check. 👍